File: app/src/main/java/com/mnj/java_solutions/BalancedStringLC1221.java
package com.mnj.java_solutions;

public class BalancedStringLC1221 {

    public void balancedStringSplit() {
        String str = "RLRRLLRLRL";

        int result = 0;

        int balance = 0;

        for (char c : str.toCharArray()) {
            if (c == 'R')
                balance++;
            else if (c == 'L')
                balance--;

            if (balance == 0)
                result++;
        }
        System.out.println("==>> maximum balanced substring are  :" + result);
    }
}

File: app/src/main/java/com/mnj/java_solutions/BalancedStringLC1221.kt
package com.mnj.java_solutions

class BalancedStringLC1221 {
    fun balancedStringSplit() {
        val str = "RLRRLLRLRL"
        var result = 0
        var balance = 0
        for (c in str.toCharArray()) {
            if (c == 'R') balance++ else if (c == 'L') balance--
            if (balance == 0) result++
        }
        println("==>> maximum balanced substring are  :$result")
    }
}